trace/Span.md

```diff
@@ -24,9 +27,54 @@ parent span id for the child Span.
 #### TraceOptions 
 Represents the options for a trace. It is represented as 1 byte (bitmap).
 
-##### Supported bits
+##### Supported Bits
 * Sampling bit -  Bit to represent whether trace is sampled or not (mask `0x1`).
 
-## Span creation API
+## Span Creation
+The implementation MUST allow users to create two types of Spans:
+* Root Spans - spans that do not have a parent.
+* Child Spans - the parent can be explicitly set or inherit from the Context.
 
-TODO(bdrutu): Add details here.
\ No newline at end of file
+When creating a Span the implementation MUST allow users to create the Span attached or detached 
+from the Context, this allows users to manage the interaction with the Context independently of 
+the Span lifetime:
+* Attached to the Context - the newly created Span is attached to the Context.
+* Detached from the Context - the newly created Span is not attached to any Context.
+
+### How Span interacts with Context?
+Context interaction represents the process of attaching/detaching a Span to the Context 
+in order to propagate it in-process (possibly between threads) and between function calls.
+
+There are two supported implementations for the Context based on how the propagation is implemented:
+* With implicit propagation - implicitly passed between function calls and threads, usually 
+implemented using thread-local variables (e.g. Java [io.grpc.Context][javaContext])
+* With explicit propagation - explicitly passed between function calls and threads (e.g. Go 
+[Context][goContext])
+
+When an implicit propagated Context is used implementation MUST use scoped objects to 
+attach/detach a Span (scoped objects represent auto closable objects, e.g. stack allocated 
+objects in C++):
+* When attach/detach an already created Span the API MAY be called `WithSpan`.
+* When attach/detach at the creation time the API MAY be called `StartSpan` or `StartScopedSpan`.
+
+When an explicit propagated Context is used implementation MUST create a new Context when a Span 
+is attached (immutable Context):
+* When attach/detach an already created Span the API MAY be called `WithSpan`.
+* When attach/detach at the creation time the API MAY be called `StartSpan` or `StartScopedSpan`.
+
+### What is Span Lifetime?
+Span lifetime represents the process of recording the start and the end timestamps to the Span 
+object:
+* The start time is recorded when the Span is created.
+* The end time needs to be recorded when the operation is ended.
```

Comment:
Add clarification that startScopedSpan does imply closing the scope finishes the lifetime. Not sure it is made obvious yet